File: 卡码/代码随想录/1/06-二叉树/C++/02-二叉树属性/LeetCode_0257.cpp
//257. 二叉树的所有路径
#include <iostream>
#include <vector>
#include <string>
#include "../TreeNode.h"
using namespace std;

class Solution {
public:
	void traversal(TreeNode* root,string path,vector<string>& res){
		path+=to_string(root->val);
		//到达叶子节点
		if(!root->left&&!root->right){
			res.push_back(path);
			return;
		}
		if(root->left) traversal(root->left,path+"->",res);
		if(root->right) traversal(root->right,path+"->",res);
	}
  vector<string> binaryTreePaths(TreeNode* root) {
		vector<string> res;
		string path;
		if(!root) return res;
		traversal(root,path,res);
		return res;
  }
};

int main(){
  system("pause");
  return 0;
}

File: 卡码/代码随想录/1/06-二叉树/Java/binaryTree/src/main/java/com/kurumi/binarytree/_02_二叉树属性/LeetCode_0257.java
//257. 二叉树的所有路径
package com.kurumi.binarytree._02_二叉树属性;

import com.kurumi.binarytree.TreeNode;
import java.util.ArrayList;
import java.util.List;

public class LeetCode_0257 {

}
class Solution {
  public void traversal(TreeNode root,String path,List<String> res){
    path+=root.val;
    //到达叶子节点
    if(root.left==null&&root.right==null){
      res.add(path.toString());
      return;
    }
    if(root.left!=null) traversal(root.left, path+"->", res);
    if(root.right!=null) traversal(root.right,path+"->",res);
  }
  public List<String> binaryTreePaths(TreeNode root) {
    List<String> res=new ArrayList<>();
    String path="";
    if(root==null) return res;
    traversal(root, path, res);
    return res;
  }
}
